frontend/viewer/src/lib/admin-dialogs/GetProjectByCodeDialog.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+  import {t} from 'svelte-i18n-lingui';
+  import {Button} from '$lib/components/ui/button';
+  import * as Dialog from '$lib/components/ui/dialog';
+  import {useBackHandler} from '$lib/utils/back-handler.svelte';
+  import Input from '$lib/components/ui/input/input.svelte';
+  import Label from '$lib/components/ui/label/label.svelte';
+  import {ProjectRole} from '$lib/dotnet-types/generated-types/LexCore/Entities/ProjectRole';
+  import Select from '$lib/components/field-editors/select.svelte';
+
+  let open = $state(false);
+  let loading = $state(false);
+  let error: string|undefined = $state();
+  useBackHandler({addToStack: () => open, onBack: () => open = false, key: 'get-project-by-code-dialog'});
+
+  let { onDownloadProject, validateCode }: {
+    onDownloadProject: (code: string, userRole: ProjectRole) => Promise<string | undefined>,
+    validateCode: (code: string) => string | undefined,
+  } = $props();
+
+  async function downloadProject(e: Event, projectCode: string, userRole: ProjectRole) {
+    e.preventDefault();
+    e.stopPropagation();
+    loading = true;
+    error = await onDownloadProject(projectCode, userRole);
+    loading = false;
+    open = !!error;
+  }
+
+  let projectCode = $state('');
+  let codeValidationError = $derived(validateCode(projectCode));
+  let userRole: ProjectRole = $state(ProjectRole.Observer);
+  const validRoles = Object.keys(ProjectRole).filter((role) => (role as keyof typeof ProjectRole) !== 'Unknown');
+
+  export function openDialog()
+  {
+    error = undefined;
+    projectCode = '';
+    userRole = ProjectRole.Observer;
+    loading = false;
+    open = true;
+  }
+</script>
+
+{#if open}
+<Dialog.Root bind:open={open}>
+  <Dialog.DialogContent>
+    <Dialog.DialogHeader>
+      <Dialog.DialogTitle>{$t`Download project by project code`}</Dialog.DialogTitle>
+    </Dialog.DialogHeader>
+    <Label class="cursor-pointer flex items-center gap-2">
+      Code:
+      <Input bind:value={projectCode} />
+    </Label>
+    <Label class="cursor-pointer flex items-center gap-2">
+      Role:
+      <Select
+          bind:value={userRole}
+          options={validRoles}
+          labelSelector={(role: ProjectRole) => $t(role)}
```

Comment:
this isn't going to work for i18n. In FW Lite the code is parsed and the text extracted. I suggest you do something like this:
```ts
const roles = [
{value: ProjectRole.Observer, label: $t`Observer`}
...
]
```
this way the i18n parser will see the string available for translation. The other option would be to just use a hard coded radio group, not sure if I like the dropdown select or radio group UI better, up to you.

Reply:
Made a hardcoded list of roles in commit d5eeaf42d, and also switched to radio buttons to see how it looks. It's meh, I might go back to Select since I kind of like the look of the dropdown better. But I'll keep the hardcoded list of roles when I go back to Select.

Reply:
Okay, commit f89d7e65f has something I'm reasonably happy with code-wise, given the constraint on labels being hardcoded as strings somewhere in the code. I switched back to Select as I just like the look better than radio buttons. The whole `const observerSelection = roleSelections[2];` thing feels just a little silly, but it felt even sillier to do `const observerSelection = roleSelections.find(r => r.role === UserProjectRole.Observer)!;` since the list is *right there* one line above.

This is done now unless there's something about it that you want changed.